### ISSUE TITLE
data set

### ISSUE THREAD

**Issue by zi1zi** [clovaai/lffont#38]
Hello, I have the following questions:
1. Does your model train multiple fonts at one time?
2. Is the dataset.json file in the example to generate train.json and test.json
3. Does the valid field in train.json represent the validation set? Why are there seen_fonts, seen_units, unseen_fonts, unseen_units? Among them, is unseen_fonts a type of font that has not been seen during training? If so, how does the model generate characters in font classes it has not seen before at evaluation time?
4. https://github.com/clovaai/fewshot-font-generation This preparation data set is much simpler than LF-font, is there any difference

**Comment by SanghyukChun** [comment]
1. Yes.
2. I cannot get your question well, but the json files are in https://github.com/clovaai/lffont/tree/master/meta If you are asking that https://github.com/clovaai/lffont/blob/master/meta/dataset.json is the same as the our model in the paper, no that is an example. You should make your own one.
3. They are a validation set, not seen during the training. I will not explain how our model works here. Please read the paper first.
4. https://github.com/clovaai/fewshot-font-generation is a more updated version and is a unified version of LF-Font, MX-Font and other font generation methods.

If you want to make fonts without understanding anything, you'd better to use https://github.com/kaonashi-tyc/zi2zi
If you are considering to use the LF-Font repository as your own research project, I recommend using https://github.com/clovaai/fewshot-font-generation
Otherwise, you may want to reproduce LF-Font on your own dataset, in this case, you'd better to use either this repository or `fewshot-font-generation`

**Comment by zi1zi** [comment]
Thank you for your prompt reply
My second question is that both train.json and test.json are made by build_dataset.py? That is, the json file generated by the --json_path path/to/save/dict path

**Comment by SanghyukChun** [comment]
No. `build_dataset.py` only returns one json. Read the code for the details https://github.com/clovaai/lffont/blob/master/build_dataset.py#L79-L81
You should manually divide the output json file for train and validation.

**Comment by SanghyukChun** [comment]
Closing the issue, assuming the answer resolves the problem.
Please re-open the issue as necessary.
